packages/stores/src/account/base.ts

```diff
@@ -309,6 +331,63 @@ export class AccountStore<Injects extends Record<string, any>[] = []> {
     return Boolean(wallet);
   }
 
+  connectedWalletSupportsChain(
+    chainId: string
+  ): IPromiseBasedObservable<boolean> | undefined {
+    if (!chainId) return undefined;
+
+    /**
+     * Retrieve the Osmosis chain wallet. Other wallets might not be connected
+     * due to lack of support or pending approval. However, Osmosis is always
+     * approved upon connecting to the app.
+     */
+    const wallet = this.getWallet(this.osmosisChainId);
+
+    if (!wallet || wallet.walletStatus !== WalletStatus.Connected) {
+      return undefined;
+    }
+
+    const id = `${wallet.walletName}_${chainId}`;
+
+    let promiseObservable = this._walletToSupportChainPromise.get(id);
+
+    if (!promiseObservable) {
+      promiseObservable = fromPromise<boolean>(wallet.supportsChain(chainId));
+      this._walletToSupportChainPromise.set(id, promiseObservable);
```

Comment:
Is there any scenario where we want to delete a promise? Or is it for the lifetime of the app?

Reply:
It should be for the lifetime of the app. The wallet's available chains shouldn't change unless it has the "suggest chain" feature, which wouldn't need this method